### PR TITLE
nix-template: 0.1.1 -> 0.1.4

### LINKED ISSUE #16: completions fail to load
i put 
```
eval "$(nix-template completions zsh)"
```
in my zshrc

i get `_arguments: command not found` when i start a new shell

happens without a config too
```
env -i zsh -d -f -i
eval "$(nix-template completions zsh)"
```

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
add support for pypi and shell completions

https://github.com/jonringer/nix-template/releases/

closes: https://github.com/jonringer/nix-template/issues/16

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by prusnak — APPROVED** [review]

File: pkgs/tools/package-management/nix-template/default.nix [hunk 1]
```diff
@@ -8,25 +9,35 @@
...
     ++ lib.optional stdenv.isDarwin Security;
 
   # needed for nix-prefetch-url
   postInstall = ''
     wrapProgram $out/bin/nix-template \
       --prefix PATH : ${lib.makeBinPath [ nix ]}
+
+    installShellCompletion --bash --cmd nix-template <($out/bin/nix-template completion bash)
+    installShellCompletion --zsh --cmd nix-template <($out/bin/nix-template completion zsh)
+    installShellCompletion --fish --cmd nix-template <($out/bin/nix-template completion fish)
```
Comment (Artturin) [inline pkgs/tools/package-management/nix-template/default.nix:40]:
```suggestion
    installShellCompletion --cmd nix-template \
      --bash <($out/bin/nix-template completions bash) \
      --fish <($out/bin/nix-template completions fish) \
      --zsh <($out/bin/nix-template completions zsh)
```

Comment (jonringer) [reply pkgs/tools/package-management/nix-template/default.nix:40]: I don't know if that's strictly better.

Had issues trying to use installShellCompletion in the past, and would rather not complicate the cli usage. Then again, I used to not be able to do a redirect, so maybe now it's better

Comment (Artturin) [reply pkgs/tools/package-management/nix-template/default.nix:40]: i tried it and it works 
if you decide to keep the current on the you have to change `..template completion` to `..template completions`

Comment (jonringer) [reply pkgs/tools/package-management/nix-template/default.nix:40]: oh, oops! Darn s :)

Comment (jonringer) [reply pkgs/tools/package-management/nix-template/default.nix]: I'll keep it for now, but thanks for the correction :)

**Review by Artturin — APPROVED** [review]

File: pkgs/tools/package-management/nix-template/default.nix [hunk 2]
```diff
@@ -8,25 +9,35 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "nix-template";
-  version = "0.1.1";
+  version = "0.1.4";
 
   src = fetchFromGitHub {
+    name = "${pname}-${version}-src";
```
Comment (Artturin) [inline pkgs/tools/package-management/nix-template/default.nix:15]: why this?

Comment (jonringer) [reply pkgs/tools/package-management/nix-template/default.nix:15]: fetchFromGitHub will name everything as just "source", but passing version number, you're guarenteed to get a different sha256 and avoid stale FOD's

File: pkgs/tools/package-management/nix-template/default.nix [hunk 3]
```diff
@@ -8,25 +9,35 @@
...
     ++ lib.optional stdenv.isDarwin Security;
 
   # needed for nix-prefetch-url
   postInstall = ''
     wrapProgram $out/bin/nix-template \
       --prefix PATH : ${lib.makeBinPath [ nix ]}
+
+    installShellCompletion --bash --cmd nix-template <($out/bin/nix-template completions bash)
+    installShellCompletion --zsh --cmd nix-template <($out/bin/nix-template completions zsh)
+    installShellCompletion --fish --cmd nix-template <($out/bin/nix-template completions fish)
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/nix-template/default.nix:40]:
```suggestion
    installShellCompletion --cmd nix-template \
      --bash <($out/bin/nix-template completions bash) \
      --fish <($out/bin/nix-template completions fish) \
      --zsh <($out/bin/nix-template completions zsh)
```

**Pull request merged**